php_phongo.c

```diff
@@ -1388,6 +1388,20 @@ static bool php_phongo_apply_options_to_uri(mongoc_uri_t *uri, bson_t *options T
 
 			continue;
 		}
+
+		if (!strcasecmp(key, MONGOC_URI_COMPRESSORS)) {
+			if (!BSON_ITER_HOLDS_UTF8(&iter)) {
+				PHONGO_URI_INVALID_TYPE(iter, "string");
+				return false;
+			}
+
+			if (!mongoc_uri_set_compressors(uri, bson_iter_utf8(&iter, NULL))) {
```

Comment:
Can we get some error tests?

You should be able to copy existing tests for `PHONGO_URI_INVALID_TYPE()` errors. For `mongoc_uri_set_compressors()`, I see it only returns false if the value is `NULL` (not possible with this code path) or the string contains invalid UTF-8. There should be some invalid UTF-8 strings in other tests (e.g. `bson-toPHP_error-004.phpt`) that you can re-use.

AFAIK, there's no way to determine if compressors are enabled once the client is constructed, so we probably don't need tests to demonstrate successful application of the option. If you did, though, it looks like "noop" is always a supported compressor type (even if snappy and zlib are disabled).